tests/test_smoke.py

```diff
@@ -2434,6 +2434,44 @@ def test_new_bucket_creation_and_deletion(self, tmp_local_storage_obj,
         out = subprocess.check_output(['sky', 'storage', 'ls'])
         assert tmp_local_storage_obj.name not in out.decode('utf-8')
 
+    @pytest.mark.parametrize('store_type', [
+        storage_lib.StoreType.S3, storage_lib.StoreType.GCS,
+        pytest.param(storage_lib.StoreType.R2, marks=pytest.mark.cloudflare)
+    ])
+    def test_multiple_buckets_creation_and_deletion(self, store_type):
+        # Creates multiple new buckets(5 buckets) with a local source, uploads
+        # files and deletes them.
+        storage_obj_name = []
+        for _ in range(5):
+            timestamp = str(time.time()).replace('.', '')
+            with tempfile.TemporaryDirectory() as tmpdir:
+                subprocess.check_output(f'mkdir -p {tmpdir}', shell=True)
+                subprocess.check_output(f'touch {tmpdir}/test-{timestamp}.txt',
+                                        shell=True)
+                store_obj = storage_lib.Storage(name=f'sky-test-{timestamp}',
+                                                source=tmpdir)
+                store_obj.add_store(store_type)
+                storage_obj_name.append(store_obj.name)
```

Comment:
We should move this storage object creation to a fixture to avoid leaks in the event this test fails. E.g., something like`def tmp_multiple_scratch_storage_obj`, which looks like `tmp_scratch_storage_obj` except we yield a list of 5 randomly generated storages. Then we can call add_store on each of them in this test.

Reply:
We also probably don't need to upload files - empty storage would be fine for this test (see `tmp_scratch_storage_obj`).

Reply:
@romilbhardwaj, I moved the test to a fixture and removed the file upload part during storage creation.